lib/utils.js

```diff
@@ -33,7 +33,8 @@ function debrack (s) {
 }
 
 function uriToFilename (uri, base) {
-  var filename = path.join(base, uri)
+  var decoded = uri.split('/').map(decodeURIComponent).join('/')
```

Comment:
A goal of this project is to be able to export existing files system users have. This means hex encoding things like space which are not allowed in URLs.   A Windows or Mac Documents or Music folder should be be able to export into Solid in a useful way.

It is NOT a goal to be able to allow clients to put files with characters which are typically illegal in a file system, like say colon, slash, or vertical bar.   I think we could return an error when a PUT or  PATCH tries to do that (by  hex encoding them in a URL).  I don't see any value in allowing this characters.